klient/k8s/resources/decoder.go

```diff
@@ -0,0 +1,354 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resources
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/apimachinery/pkg/util/yaml"
+	"k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/e2e-framework/klient/k8s"
+)
+
+func isDirectory(dir string) bool {
+	d, err := os.Stat(dir)
+	if err != nil {
+		return false
+	}
+	return d.IsDir()
+}
+
+func listResourceFiles(directory string) ([]string, error) {
+	// determine if directory is a path to a directory, or if it has a filename Glob included
+	var base string
+	dir, err := os.Stat(directory)
+	if err != nil { // if this is not a resolvable path, assume the "base" of the path is a filename-prefix
+		base = filepath.Base(directory)
```

Comment:
The `err` should be handled with more safeguard. For instance, it could be non-existent path,  a permission error, etc. Check the error first, before continuing.